### PR TITLE
Docs 4476

### PR DESCRIPTION
**Pull request by george-hoffer**
https://www.mulesoft.org/jira/browse/DOCS-4476 -- Cleaned up all the procedures and put code examples in the order one sees in Studio.

### CODE REVIEW

File: modules/ROOT/pages/http/http-authentication.adoc [hunk 1]
```diff
@@ -2,357 +2,496 @@
...
 
-Use the authentication that you configure in HTTP requests when your Mule app is sending requests to a service that requires authentication, such as the Github OAuth2 server described in <<OAuth2 - Authorization Code>>. In this case, your Mule app is the client. On the other hand, when you want to protect your API or app from _receiving requests_ from unauthorized access, use an API Manager policy, such as the xref:api-manager::external-oauth-2.0-token-validation-policy.adoc[OAuth 2.0 Access Token Enforcement Using External Provider] policy.
+Use the authentication that you configure in HTTP requests when your Mule runtime (Mule) app is sending requests to a service that requires authentication, such as the GitHub OAuth2 server described in <<OAuth2 - Authorization Code>>. In this case, your Mule app is the client. On the other hand, when you want to protect your API or app from receiving requests from unauthorized access, use an API Manager policy, such as the xref:api-manager::external-oauth-2.0-token-validation-policy.adoc[OAuth 2.0 Access Token Enforcement Using External Provider] policy.
```
Comment (elrodro83) [inline modules/ROOT/pages/http/http-authentication.adoc]: `app is sending requests to a service that requires authentication` -> `app uses the <HTTP Connector> to send requests to a service that requires authentication`

File: modules/ROOT/pages/http/http-authentication.adoc [hunk 2]
```diff
@@ -2,357 +2,496 @@
...
+
+In addition, in Studio you can add an OAuth component's operation to the canvas, which 
+automatically adds the following dependency to the project's pom.xml file:
+
+[source,xml,linenums]
+----
+<dependency>
+    <groupId>org.mule.modules</groupId>
+    <artifactId>mule-oauth-module</artifactId>
+    <version>1.1.2</version>
```
Comment (elrodro83) [inline modules/ROOT/pages/http/http-authentication.adoc]: latest version is 1.1.8

File: modules/ROOT/pages/http/http-authentication.adoc [hunk 3]
```diff
@@ -2,357 +2,496 @@
...
 . Save changes.
 +
 The error indicator disappears.
 
-*Next, set up an HTTP Request connector:*
+=== Basic Authentication - Configure an HTTP Request Connector
 
-. Drag another HTTP connector from the palette, and drop it in the Process area of the flow.
-. In the properties editor, in Connector Configuration, click image:http/add-16x16.png[Add-16x16].
+. Drag an *HTTP* > *Request* operation from the Mule Palette to the Process area of the Studio flow. This operation is known as the HTTP Request connector.
```
Comment (elrodro83) [inline modules/ROOT/pages/http/http-authentication.adoc]: Remove `This operation is known as the HTTP Request connector`. Instead, add a link to the http request operation documentation page

File: modules/ROOT/pages/http/http-authentication.adoc [hunk 4]
```diff
@@ -2,357 +2,496 @@
...
-* Client Id: Type the client Id that Github provided when you registered the app.
-* Client Secret: Type the client secret that Github provided when you registered the app.
-* Local callback URL: `+http://localhost:8082/callback+`
+[[setpomfile]]
+[source,xml,linenums]
+----
+<dependency>
+    <groupId>org.mule.modules</groupId>
+    <artifactId>mule-oauth-module</artifactId>
+    <version>1.1.2</version>
```
Comment (elrodro83) [inline modules/ROOT/pages/http/http-authentication.adoc]: 1.1.8

File: modules/ROOT/pages/http/http-authentication.adoc [hunk 5]
```diff
@@ -2,357 +2,496 @@
...
+    <classifier>mule-plugin</classifier>
+</dependency>
+----
++
+. In the properties editor for *Connector Configuration*, click the green plus icon (or click the Edit icon if you have already created a configuration).
+. Set the *Authentication* field to *Authorization code grant type*.
+. Set the following required fields:
++
+** *External callback url*: `+http://localhost:8082/callback+`
```
Comment (elrodro83) [inline modules/ROOT/pages/http/http-authentication.adoc]: `localhost` is not a good example for an external url. Change it to `myapp.mycompany.com` or something along those lines

File: modules/ROOT/pages/http/http-authentication.adoc [hunk 6]
```diff
@@ -2,357 +2,496 @@
...
+You can also set these optional fields:
 +
-This value matches the value you configured for *Authorization callback URL* when registering the app in Github.
-* Authorization Url: `+https://github.com/login/oauth/authorize+`
-* Local Authorization Url: `+https://localhost:8082/login+`
-* Token Url: `+https://github.com/login/oauth/access_token+`
-* Response Access Token: `#[payload.'access_token']`
+** *Local callback url*: `+http://localhost:8082/callback+`
++
+This value matches the value you configured for *Authorization callback URL* when registering the app in GitHub.
```
Comment (elrodro83) [inline modules/ROOT/pages/http/http-authentication.adoc]: Authorization callback URL -> External callback url

Comment (elrodro83) [reply modules/ROOT/pages/http/http-authentication.adoc]: This is the configuration of the server that mule will create to receive the requests that a remote host sends to the externalCallbackUrl. external and internal callback url are the same, just that one is the configuration to create a server in the runtime (internal) and the other is how that server is seen from the internet (external). This should be explained here in the docs.

File: modules/ROOT/pages/http/http-authentication.adoc [hunk 7]
```diff
@@ -2,357 +2,496 @@
...
-            clientId="27...df" clientSecret="ae...6"
-            localCallbackUrl="http://localhost:8082/callback"
-        	   authorizationUrl="https://github.com/login/oauth/authorize"
-        	   localAuthorizationUrl="http://localhost:8082/login"
-            tokenUrl="https://github.com/login/oauth/access_token"
-        	   responseAccessToken="#[payload.'access_token']"
-        	   responseRefreshToken="#[payload.'access_token']">
-        </oauth:authorization-code-grant-type>
-    </http:request-config>
+        externalCallbackUrl="http://localhost:8082/callback"
```
Comment (elrodro83) [inline modules/ROOT/pages/http/http-authentication.adoc]: previous comment about external callback applies here as well

File: modules/ROOT/pages/http/http-authentication.adoc [hunk 8]
```diff
@@ -386,264 +526,316 @@ The Github API returns your user information.
...
-=== Studio Visual Editor
+=== OAuth 2 Custom Parameters - Studio Visual Editor
 
 . In Studio, select the HTTP Request Configuration global element where you want to use the OAuth authorization code grant type.
-. On the *Authentication* combo box, select `Authorization code grant type`.
-. Fill in the same fields as in the previous example.
+. In *Authentication*, select *Authorization code grant type*. If this option is not visible, <<setpomfile,set the pom.xml file>> for OAuth options.
+. Fill in the same fields as in the previous example:
++
+** External Callback URL = `+http://localhost:8082/callback+`
```
Comment (elrodro83) [inline modules/ROOT/pages/http/http-authentication.adoc]: previous comment about external callback applies here as well

File: modules/ROOT/pages/http/http-authentication.adoc [hunk 9]
```diff
@@ -386,264 +526,316 @@ The Github API returns your user information.
...
-            tokenUrl="http://www.box.com/api/oauth2/token">
-                <oauth:custom-parameters>
-                    <oauth:custom-parameter
-                        key="box_device_id" value="123142"/>
-                    <oauth:custom-parameter
-                        key="box_device_name" value="my-phone"/>
-                </oauth:custom-parameters>
-        </oauth:authorization-code-grant-type>
-    </http:request-config>
+        externalCallbackUrl="http://localhost:8082/callback"
```
Comment (elrodro83) [inline modules/ROOT/pages/http/http-authentication.adoc]: previous comment about external callback applies here as well

File: modules/ROOT/pages/http/http-authentication.adoc [hunk 10]
```diff
@@ -386,264 +526,316 @@ The Github API returns your user information.
...
-            responseAccessToken="#[payload.'access_token']"
-            responseExpiresIn="#[payload.'expires_in']"
-             responseRefreshToken="#[payload.'refresh_token']>
-        </oauth:authorization-code-grant-type>
-    </http:request-config>
+                   host="api.box.com" port="443" basePath="/2.0">
+      <http:authentication>
+          <oauth:authorization-code-grant-type
+              localCallbackUrl="http://localhost:8082/redirectUrl"
+              externalCallbackUrl="http://localhost:8082/callback"
```
Comment (elrodro83) [inline modules/ROOT/pages/http/http-authentication.adoc]: previous comment about external callback applies here as well

File: modules/ROOT/pages/http/http-authentication.adoc [hunk 11]
```diff
@@ -386,264 +526,316 @@ The Github API returns your user information.
...
-            clientId="your_client_id" clientSecret="your_client_secret"
-            localCallbackUrl="http://localhost:8082/redirectUrl"
-            authorizationUrl="http://www.box.com/api/oauth2/authorize"
-            localAuthorizationUrl="http://localhost:8082/authorization"
-            tokenUrl="http://www.box.com/api/oauth2/token"
-            refreshTokenWhen="#[payload.response.status == 'unauthorized']">
-        </oauth:authorization-code-grant-type>
-    </http:request-config>
+        localCallbackUrl="http://localhost:8082/redirectUrl"
+        externalCallbackUrl="http://localhost:8082/callback"
```
Comment (elrodro83) [inline modules/ROOT/pages/http/http-authentication.adoc]: previous comment about external callback applies here as well

File: modules/ROOT/pages/http/http-authentication.adoc [hunk 12]
```diff
@@ -386,264 +526,316 @@ The Github API returns your user information.
...
 <http:request-config name="HTTP_Request_Configuration"
-                     host="api.box.com" port="443" basePath="/2.0"
-                     tlsContext-ref="TLS_Context">
+         host="api.box.com" port="443" basePath="/2.0"
+         tlsContext-ref="TLS_Context">
+    <http:authentication>
         <oauth:authorization-code-grant-type
-            clientId="your_client_id" clientSecret="your_client_secret"
             localCallbackUrl="http://localhost:8082/redirectUrl"
+            externalCallbackUrl="http://localhost:8082/callback"
```
Comment (elrodro83) [inline modules/ROOT/pages/http/http-authentication.adoc]: previous comment about external callback applies here as well

File: modules/ROOT/pages/http/http-authentication.adoc [hunk 13]
```diff
@@ -386,264 +526,316 @@ The Github API returns your user information.
...
-                     tlsContext-ref="TLS_Context" protocol="HTTPS">
+         host="api.box.com" port="443" basePath="/2.0"
+         tlsContext-ref="TLS_Context" protocol="HTTPS">
+    <http:authentication>
         <oauth:authorization-code-grant-type
-            clientId="your_client_id" clientSecret="your_client_secret"
             localCallbackUrl="http://localhost:8082/redirectUrl"
-            tlsContext="TLS_Context"
-            authorizationUrl="https://www.box.com/api/oauth2/authorize"
+            externalCallbackUrl="http://localhost:8082/callback"
```
Comment (elrodro83) [inline modules/ROOT/pages/http/http-authentication.adoc]: previous comment about external callback applies here as well

**Review by elrodro83 — CHANGES_REQUESTED** [review]

**Review by afelisatti — APPROVED** [review]

File: modules/ROOT/pages/http/http-authentication.adoc (hunk 5 above)
Comment (george-hoffer) [reply modules/ROOT/pages/http/http-authentication.adoc]: Fixed globally

File: modules/ROOT/pages/http/http-authentication.adoc (hunk 4 above)
Comment (george-hoffer) [reply modules/ROOT/pages/http/http-authentication.adoc]: fixed

File: modules/ROOT/pages/http/http-authentication.adoc (hunk 13 above)
Comment (george-hoffer) [reply modules/ROOT/pages/http/http-authentication.adoc]: fixed

File: modules/ROOT/pages/http/http-authentication.adoc (hunk 6 above)
Comment (george-hoffer) [reply modules/ROOT/pages/http/http-authentication.adoc]: Fixed

**Review by elrodro83 — APPROVED** [review]

**Pull request merged**